.github/workflows/hyper_threading_benchmarks.yml

```diff
@@ -0,0 +1,123 @@
+name: Benchmark Hyper Threading
+
+on:
+  pull_request:
+    branches:
+      - main
+
+jobs:
+  benchmark:
+    runs-on: ubuntu-latest
+    steps:
+    - name: Checkout PR
+      uses: actions/checkout@v2
+      with:
+        ref: ${{ github.head_ref }}
+
+    - name: Setup Python
+      uses: actions/setup-python@v4
+      with:
+        python-version: '3.9'
+
+    - name: Install Dependencies
+      run: |
+        pip install -r requirements.txt
+        sudo apt update
+        sudo apt-get install -y hyperfine
+
+    - name: Install Rust
+      uses: dtolnay/rust-toolchain@1.74.1
+      with:
+        components: rustfmt, clippy
+
+    - name: Compile PR Version
+      run: |
+        cargo build --release -p hyper_threading
+        cp target/release/hyper_threading ${{ github.workspace }}/hyper_threading_pr
+        cp ./examples/hyper_threading/hyper-threading-workflow.sh ${{ github.workspace }}/hyper-threading-workflow.sh
+    
+    - name: Upload PR Binary
+      uses: actions/upload-artifact@v4
+      with:
+        name: hyper_threading_pr_binary
+        path: ${{ github.workspace }}/hyper_threading_pr
+
+    - name: Upload Workflow Script
+      uses: actions/upload-artifact@v4
+      with:
+        name: hyper_threading_workflow_script
+        path: ${{ github.workspace }}/hyper-threading-workflow.sh
+
+
+    - name: Checkout Main Branch
+      uses: actions/checkout@v2
+      with:
+        ref: 'main'
+
+    - name: Compile Main Version
+      run: |
+        cargo build --release -p hyper_threading
+        cp target/release/hyper_threading ${{ github.workspace }}/hyper_threading_main
+
+    - name: Upload Main Binary
+      uses: actions/upload-artifact@v4
+      with:
+        name: hyper_threading_main_binary
+        path: ${{ github.workspace }}/hyper_threading_main
+
+    - name: Download hyper_threading_pr_binary
+      uses: actions/download-artifact@v4
+      with:
+        name: hyper_threading_pr_binary
+        path: ${{ github.workspace }}/
+
+    - name: Download hyper_threading_workflow_script
+      uses: actions/download-artifact@v4
+      with:
+        name: hyper_threading_workflow_script
+        path: ${{ github.workspace }}/
+
+    - name: Download hyper_threading_main_binary
+      uses: actions/download-artifact@v4
+      with:
+        name: hyper_threading_main_binary
+        path: ${{ github.workspace }}/
```

Comment:
This is unnecessary. Since you're running everything in a single job you still have the files where you copied them. This would only be necessary if we did parallel jobs here.

Reply:
Thank @Oppen !!
Since we are doing a git checkout main (To create the other binary and compare it vs the branch one) I still need to download and upload the PR binary and the workflow.sh. But I was able to remove the download and upload of the main binary 
https://github.com/lambdaclass/cairo-vm/pull/1711/commits/0726f23e5abeccb49a2ee0e01d8f3de69c3345ba

Reply:
Right. We may be able to remove that too in a later PR, when this is in main.

Reply:
Hold on. We can make the copy to the workspace before checking out main. That should solve it.